acra/configuring-maintaining/logging/output.md

```diff
@@ -0,0 +1,60 @@
+time="2021-07-01T19:34:14+03:00" level=info msg="Initialized default env ACRA_MASTER_KEY loader"
+time="2021-07-01T19:34:14+03:00" level=info msg="Initialising keystore..."
+time="2021-07-01T19:34:14+03:00" level=info msg="Keystore init OK"
+time="2021-07-01T19:34:14+03:00" level=info msg="Configuring transport..."
+```
+
+### JSON:
+
+JSON log contains many useful fields like: `product` and `versions`, which define Acra components; `timestamp` in RFC3339 and `unixTime` to be compatible with a variety of log analysis tools.<br/>
+The log `levels` could be `debug/info/warning/error/panic`.<br/>
+Acra logs both error code and message if an error occurs (`code` and `error` fields respectively).
+
+```
+{"level":"info","msg":"Starting service acra-server [pid=304386]","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"level":"info","msg":"Validating service configuration...","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"level":"info","msg":"Load encryptor configuration from ./acra-conf.yaml ...","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"level":"info","msg":"Encryptor configuration loaded","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"level":"info","msg":"Initializing ACRA_MASTER_KEY loader...","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"level":"info","msg":"Initialized default env ACRA_MASTER_KEY loader","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"level":"info","msg":"Initialising keystore...","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"level":"info","msg":"Keystore init OK","product":"acra-server","timestamp":"2021-07-01T19:31:17+03:00","unixTime":"1625157077.908","version":"0.85.0"}
+{"code":563,"error":"fail to parse specified query","level":"warning","msg":"Failed to parse input query","product":"acra-server","timestamp":"2021-07-01T19:47:06+03:00","unixTime":"1625158026.944","version":"0.85.0"}
+```
+
+### CEF:
+
+CEF logger is compliant with the [CEF protocol](https://kc.mcafee.com/resources/sites/MCAFEE/content/live/CORP_KNOWLEDGEBASE/78000/KB78712/en_US/CEF_White_Paper_20100722.pdf) revision 16 from July 22, 2010, except the fact that Acra is using custom Dictionary Extension fields (like `unixTime`) because we found them to be very useful for log analysis.<br/>
+Pay attention that severity levels range from 0 (debug level) to 10 (panic level), see severity mapping function in [logging/cef_formatter](https://github.com/cossacklabs/acra-Q12021/blob/master/logging/cef_formatter.go#L199).
```

Comment:
More readability. less `<br/>`, just inline the mapping instead of being lazy.

```markdown
Acra logs are compliant with the [CEF protocol](https://kc.mcafee.com/resources/sites/MCAFEE/content/live/CORP_KNOWLEDGEBASE/78000/KB78712/en_US/CEF_White_Paper_20100722.pdf) revision 16 from July 22, 2010,
except that Acra is using custom Dictionary Extension fields (like `unixTime`)
as we found them to be very useful for log analysis.

Note that CEF severity levels range from 0 to 10.
They map to Acra log levels as follows:

| CEF | Acra log level |
| --- | -------------- |
|  0  | `debug`        |
|  1  | `info`         |
|  3  | `warning`      |
|  6  | `error`        |
|  8  | `fatal`        |
| 10  | `panic`        |
```